include/range/v3/algorithm/shuffle.hpp

```diff
@@ -76,6 +76,16 @@ namespace ranges
                 return orig;
             }
 
+            template<typename I, typename S,
+                CONCEPT_REQUIRES_(RandomAccessIterator<I>() && IteratorRange<I, S>() &&
+                    Permutable<I>())>
+            I operator()(I begin, S end_) const
+            {
+                std::random_device rng;
+                std::mt19937 urng(rng());
```

Comment:
> OK, I'll add `thread_local` later tonight. Do you want to select `mt19937_64` when targetting 64-bit builds?

Should probably be `RANGES_STATIC_THREAD_LOCAL`; I don't know why the macro exists, but I assume there's a good reason.

In my (limited) experience `mt19937_64` is just slightly more efficient than `mt19937` on 64 bit implementations,
but I haven't seen hard numbers to recommend either choice - so put me down as "neutral."